CI build failure for tests/drivers/build_all/sensor

### Describe the bug

PR https://github.com/zephyrproject-rtos/zephyr/pull/90286 has introduced a regression with commit

```
commit 93a30e7806ec85713c2b126cb9a6566f4eeb4002
Author: Armando Visconti <armando.visconti@st.com>
Date:   Tue Jul 29 10:38:13 2025 +0200

    drivers/sensor/st: add support to IIS3DWB accel

    The IIS3DWB is a system-in-package featuring a 3-axis digital vibration
    sensor with low noise over an ultrawide and flat frequency range.
    The wide bandwidth, low noise, very stable and repeatable sensitivity,
    together with the capability of operating over an extended temperature
    range (up to +105 C), make the device particularly suitable for vibration
    monitoring in industrial applications.

    Datasheet: https://www.st.com/en/mems-and-sensors/iis3dwb.html

    This driver is currently only supporting the polling-mode read_and_decode
    APIs (both blocking and non-blocking).

    This driver is based on stmemsc HAL i/f v2.9.1.

    Signed-off-by: Armando Visconti <armando.visconti@st.com>
```

### Regression

- [x] This is a regression.

### Steps to reproduce

```shell
$ west build -p -b native_sim/native tests/drivers/build_all/sensor -T drivers.sensor.trigger.own.build -- -DCONFIG_COMPILER_WARNINGS_AS_ERRORS=y
```

### Relevant log output

```shell
I/__w/zephyr/modules/hal/wurthelektronik/sensors/WeSensorsSDK/WSEN_ISDS_2536030320001_hal -Wshadow -fno-strict-aliasing -Werror -Os -imacros /__w/zephyr/zephyr/twister-out/native_sim_native/host/tests/drivers/build_all/sensor/drivers.sensor.generic_test/zephyr/include/generated/zephyr/autoconf.h -fno-common -g -gdwarf-4 -fdiagnostics-color=always -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wdouble-promotion -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/__w/zephyr/zephyr/tests/drivers/build_all/sensor=CMAKE_SOURCE_DIR -fmacro-prefix-map=/__w/zephyr/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/__w/zephyr=WEST_TOPDIR -ffunction-sections -fdata-sections -m32 -msse2 -mfpmath=sse -fvisibility=hidden -fno-freestanding -std=c11 -MD -MT zephyr/drivers/sensor/st/iis3dwb/CMakeFiles/drivers__sensor__st__iis3dwb.dir/iis3dwb.c.obj -MF zephyr/drivers/sensor/st/iis3dwb/CMakeFiles/drivers__sensor__st__iis3dwb.dir/iis3dwb.c.obj.d -o zephyr/drivers/sensor/st/iis3dwb/CMakeFiles/drivers__sensor__st__iis3dwb.dir/iis3dwb.c.obj -c /__w/zephyr/zephyr/drivers/sensor/st/iis3dwb/iis3dwb.c
/__w/zephyr/zephyr/drivers/sensor/st/iis3dwb/iis3dwb.c: In function ‘iis3dwb_submit_one_shot’:
/__w/zephyr/zephyr/drivers/sensor/st/iis3dwb/iis3dwb.c:200:62: error: passing argument 7 of ‘rtio_read_regs_async’ from incompatible pointer type [-Werror=incompatible-pointer-types]
  200 |                                              iodev_sqe, dev, iis3dwb_one_shot_complete_cb);
      |                                                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                                              |
      |                                                              void (*)(struct rtio *, const struct rtio_sqe *, void *)
In file included from /__w/zephyr/zephyr/drivers/sensor/st/iis3dwb/iis3dwb.h:22,
                 from /__w/zephyr/zephyr/drivers/sensor/st/iis3dwb/iis3dwb.c:19:
/__w/zephyr/zephyr/include/zephyr/rtio/regmap.h:115:57: note: expected ‘rtio_callback_t’ {aka ‘void (*)(struct rtio *, const struct rtio_sqe *, int,  void *)’} but argument is of type ‘void (*)(struct rtio *, const struct rtio_sqe *, void *)’
  115 |                                         rtio_callback_t complete_op_cb)
      |                                         ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~
cc1: all warnings being treated as errors
```

### Impact

Functional Limitation – Some features not working as expected, but system usable.

### Environment

- OS: Ubuntu
- Zephyr SDK 0.17.4

### Additional Context

_No response_

--- a/drivers/sensor/st/iis3dwb/iis3dwb.c
+++ b/drivers/sensor/st/iis3dwb/iis3dwb.c
@@ -114,10 +114,13 @@ static int iis3dwb_attr_set(const struct device *dev, enum sensor_channel chan,
 	return 0;
 }
 
-static void iis3dwb_one_shot_complete_cb(struct rtio *ctx, const struct rtio_sqe *sqe, void *arg)
+static void iis3dwb_one_shot_complete_cb(struct rtio *ctx, const struct rtio_sqe *sqe, int result,
+					 void *arg)
 {
 	struct rtio_iodev_sqe *iodev_sqe = (struct rtio_iodev_sqe *)sqe->userdata;
 	int err = 0;
+
+	ARG_UNUSED(result);
 
 	err = rtio_flush_completion_queue(ctx);
 

--- a/drivers/sensor/st/iis3dwb/iis3dwb_stream.c
+++ b/drivers/sensor/st/iis3dwb/iis3dwb_stream.c
@@ -122,10 +122,13 @@ done:
 /*
  * Called by bus driver to complete the sqe.
  */
-static void iis3dwb_complete_op_cb(struct rtio *r, const struct rtio_sqe *sqe, void *arg)
+static void iis3dwb_complete_op_cb(struct rtio *r, const struct rtio_sqe *sqe, int result,
+				   void *arg)
 {
 	const struct device *dev = arg;
 	struct iis3dwb_data *iis3dwb = dev->data;
+
+	ARG_UNUSED(result);
 
 	/*
 	 * Mark operation completed
@@ -139,7 +142,7 @@ static void iis3dwb_complete_op_cb(struct rtio *r, const struct rtio_sqe *sqe, v
  * Called by bus driver to complete the IIS3DWB_FIFO_STATUS read op (2 bytes).
  * If FIFO threshold or FIFO full events are active it reads all FIFO entries.
  */
-static void iis3dwb_read_fifo_cb(struct rtio *r, const struct rtio_sqe *sqe, void *arg)
+static void iis3dwb_read_fifo_cb(struct rtio *r, const struct rtio_sqe *sqe, int result, void *arg)
 {
 	const struct device *dev = arg;
 	struct iis3dwb_data *iis3dwb = dev->data;
@@ -149,6 +152,8 @@ static void iis3dwb_read_fifo_cb(struct rtio *r, const struct rtio_sqe *sqe, voi
 	struct sensor_read_config *read_config;
 	uint8_t fifo_th = 0, fifo_full = 0;
 	uint16_t fifo_count;
+
+	ARG_UNUSED(result);
 
 	/* At this point, no sqe request is queued should be considered as a bug */
 	__ASSERT_NO_MSG(iis3dwb->streaming_sqe != NULL);
@@ -335,13 +340,16 @@ static void iis3dwb_read_fifo_cb(struct rtio *r, const struct rtio_sqe *sqe, voi
  * Called by bus driver to complete the IIS3DWB_STATUS_REG read op.
  * If drdy_xl is active it reads XL data (6 bytes) from IIS3DWB_OUTX_L_A reg.
  */
-static void iis3dwb_read_status_cb(struct rtio *r, const struct rtio_sqe *sqe, void *arg)
+static void iis3dwb_read_status_cb(struct rtio *r, const struct rtio_sqe *sqe, int result,
+				   void *arg)
 {
 	const struct device *dev = arg;
 	struct iis3dwb_data *iis3dwb = dev->data;
 	struct rtio *rtio = iis3dwb->rtio_ctx;
 	const struct gpio_dt_spec *irq_gpio = iis3dwb->drdy_gpio;
 	struct sensor_read_config *read_config;
+
+	ARG_UNUSED(result);
 
 	/* At this point, no sqe request is queued should be considered as a bug */
 	__ASSERT_NO_MSG(iis3dwb->streaming_sqe != NULL);

